env/work/srw_python/srwlib.py

```diff
@@ -3034,14 +3037,14 @@ def srwl_opt_setup_mask(_delta, _atten_len, _thick,
             # Hartmann hole in a rectangular shape.
             elif _grid_sh == 1:
                 # Calculate the equations for edges of rectangle.
-                xCross1 = - _grid_dx / (2 ** 0.5) * math.cos(_grid_angle)
-                yCross1 = - _grid_dx / (2 ** 0.5) * math.sin(_grid_angle)
-                xCross2 = + _grid_dx / (2 ** 0.5) * math.cos(_grid_angle)
-                yCross2 = + _grid_dx / (2 ** 0.5) * math.sin(_grid_angle)
-                k1 = math.tan(math.pi / 4 + _grid_angle)
-                k2 = -math.tan(math.pi / 4 - _grid_angle)
-                k4 = math.tan(math.pi / 4 + _grid_angle)
-                k3 = -math.tan(math.pi / 4 - _grid_angle)
+                xCross1 = - _grid_dx / (2 ** 0.5) * cos(_grid_angle)
+                yCross1 = - _grid_dx / (2 ** 0.5) * sin(_grid_angle)
+                xCross2 = + _grid_dx / (2 ** 0.5) * cos(_grid_angle)
+                yCross2 = + _grid_dx / (2 ** 0.5) * sin(_grid_angle)
+                k1 = tan(pi / 4 + _grid_angle)
+                k2 = -tan(pi / 4 - _grid_angle)
+                k4 = tan(pi / 4 + _grid_angle)
+                k3 = -tan(pi / 4 - _grid_angle)
```

Comment:
Accepted. However, note the code could be more efficient. (2**0.5),  cos(_grid_angle), sin(_grid_angle) could be calculated only once. I did these changes.